src/Margay.cpp

```diff
@@ -800,30 +817,60 @@ void Margay::resetWDT()  //Send a pulse to "feed" the watchdog timer
 	digitalWrite(WDHold, LOW);
 }
 
-void Margay::externalI2C(bool state)
+void Margay::switchExternalI2C(bool desiredState)
 {
 	/*
 	Must be ON to read off-board sensors, OFF to read on-board sensors and RTC
 	Serves to isolate these s.t. I2C addresses may not clash.
 	*/
 	pinMode(I2C_SW, OUTPUT);
-	if ( state == ON ){
+
+	if ( desiredState == ON ) {
 		digitalWrite(I2C_SW, HIGH);
+		externalI2COn = digitalRead(I2C_SW);
 	}
-	else{
+	else {
 		digitalWrite(I2C_SW, LOW);
+		externalI2COn = digitalRead(I2C_SW);
 	}
 	delay(1); // Any time needed to switch states; may not be necessary
 }
 
+void Margay::closeFarmGateI2C(bool initStateI2C)
+{
+	// This closes the farm gate at the end of a function that uses external I2C bus. 
+	// It works by by checking if the current state of I2C Comms (class variable)
+	// matches the starting state (argument to this function). If they do not match (XOR), 
+	// and it was on at the start, then turn on. Otherwise, turn off.
+	
+	if ((initStateI2C == !externalI2COn) && initStateI2C) {
+		switchExternalI2C(ON);
+	} else {
+		switchExternalI2C(OFF);
+	}
+}
+
 void Margay::addDataPoint(String (*update)(void)) //Reads new data and writes data to SD
 {
 	String data = "";
 	if(Model >= MODEL_2v0) EnviroSense.begin(0x77); //Re-initialize BME280  //FIX??
 	// Serial.println("Called Update"); //DEBUG!
-	externalI2C(ON);
-	data = (*update)(); //Run external update function
-	externalI2C(OFF);
+	
+	bool initialStateI2C = digitalRead(I2C_SW);
+
+	switchExternalI2C(ON);
+	data = (*update)(); //Run external update function	
+	switchExternalI2C(OFF);
+
+	// make sure I2C Bus is returned to initial state
+	closeFarmGateI2C(initialStateI2C);
```

Comment:
Can open or close gate, so rename:
```suggestion
	farmGateI2C(initialStateExternI2C);
```